pkg/flare/archive.go

```diff
@@ -33,6 +34,36 @@ import (
 // SearchPaths is just an alias for a map of strings
 type SearchPaths map[string]string
 
+// PermsInfos holds permissions info about the files shipped
+// in the flare.
+// It is bound to a given flare by the tempDir/hostname fields.
+// The filemode field is the file mode which will be used to create
+// the temporary permissions.log file on filesystem before being written in the zip.
+type PermsInfos struct {
+	tempDir  string
+	hostname string
+	filemode os.FileMode
+	// the key is the filepath of the file.
+	infos map[string]permsInfo
+}
+
+type permsInfo struct {
```

Comment:
nit: could we rename this into `filePermsInfo`?

Reply:
Definitely. Would you also prefer File*s*PermsInfos for the public struct containing all infos?

Reply:
I don't think you necessarily need to change the name to `FilesPermsInfos()` (though you can if you want, your decision. I didn't mention changing it because the current name leaves room for _other_ permissions maybe unrelated to files - though nothing really comes to mind now of what that could be). Regarding making it private, I think that might make sense if we're not using it anywhere. We can always make it public in the future 👍